src/irmin-fs/irmin_fs.ml

```diff
@@ -436,3 +418,9 @@ module IO_mem = struct
     Hashtbl.clear t.watches;
     Lwt.return_unit
 end
+
+(* Enforce that {!S} is a sub-type of {!Irmin.Maker}. *)
+module _ : Irmin.Maker = Maker (IO_mem)
```

Comment:
We should maybe not use anonymous modules if we want to support older versions than `ocaml.4.10`